app/controllers/portal/client_logins_controller.rb

```diff
@@ -12,25 +12,34 @@ def new
     def create
       @form = RequestClientLoginForm.new(request_client_login_params)
       if @form.valid?
-        ClientEmailLoginRequestJob.perform_later(
-          email_address: @form.email_address,
-          locale: I18n.locale,
-          visitor_id: visitor_id
-        ) if @form.email_address.present?
-
-        ClientTextMessageLoginRequestJob.perform_later(
-          sms_phone_number: @form.sms_phone_number,
-          locale: I18n.locale,
-          visitor_id: visitor_id
-        ) if @form.sms_phone_number.present?
+        if @form.email_address.present?
+          ClientEmailLoginRequestJob.perform_later(
+            email_address: @form.email_address,
+            locale: I18n.locale,
+            visitor_id: visitor_id
+          )
+          session[:email_address] = @form.email_address
+        end
+
+        if @form.sms_phone_number.present?
+          ClientTextMessageLoginRequestJob.perform_later(
+            sms_phone_number: @form.sms_phone_number,
+            locale: I18n.locale,
+            visitor_id: visitor_id
+          )
+          session[:sms_phone_number] = @form.sms_phone_number
+        end
 
         redirect_to login_link_sent_portal_client_logins_path
       else
         render :new
       end
     end
 
-    def link_sent; end
+    def link_sent
```

Comment:
This feels like a weird name for a method that's actually executing a side effect. Maybe call it "after_link_sent" to make it clear it gets called as a result of something happening?

Reply:
Since it's a controller action, I kinda think the name is OK! I'd be 100% up to spend a few min brainstorming ideas for this with you.

Reply:
Personally, I don't like `after_link_sent` since it sounds like an ActiveRecord hook.